crates/ty_python_semantic/src/types/constraints.rs

```diff
@@ -1833,6 +1976,502 @@ impl<'db> ConstraintAssignment<'db> {
     }
 }
 
+/// A collection of _sequents_ that describe how the constraints mentioned in a BDD relate to each
+/// other. These are used in several BDD operations that need to know about "derived facts" even if
+/// they are not mentioned in the BDD directly. These operations involve walking one or more paths
+/// from the root node to a terminal node. Each sequent describes paths that are invalid (which are
+/// pruned from the search), and new constraints that we can assume to be true even if we haven't
+/// seen them directly.
+///
+/// We support several kinds of sequent:
```

Comment:
Just curious about the terminology here. The special thing about the [equivalent term in logic](https://en.wikipedia.org/wiki/Sequent) seems to be that there can be multiple "consequents" on the right hand side. But none of the cases listed below seem to have that form?